src/app/SubscriptionResumptionStorage.h

```diff
@@ -0,0 +1,158 @@
+        bool mFabricFiltered;
+        Platform::ScopedMemoryBufferWithSize<AttributePathParamsValues> mAttributePaths;
+        Platform::ScopedMemoryBufferWithSize<EventPathParamsValues> mEventPaths;
+        CHIP_ERROR SetAttributePaths(const ObjectList<AttributePathParams> * pAttributePathList)
+        {
+            mAttributePaths.Free();
+            const ObjectList<AttributePathParams> * attributePath = pAttributePathList;
+            size_t attributePathCount                             = 0;
+            while (attributePath)
+            {
+                attributePathCount++;
+                attributePath = attributePath->mpNext;
+            }
+            ReturnErrorCodeIf((attributePathCount * sizeof(AttributePathParamsValues)) > UINT16_MAX, CHIP_ERROR_NO_MEMORY);
+            mAttributePaths.Calloc(attributePathCount);
+            ReturnErrorCodeIf(mAttributePaths.Get() == nullptr, CHIP_ERROR_NO_MEMORY);
+            attributePath = pAttributePathList;
+            for (size_t i = 0; i < attributePathCount; i++)
+            {
+                mAttributePaths[i].SetValues(attributePath->mValue);
+                attributePath = attributePath->mpNext;
+            }
+            return CHIP_NO_ERROR;
+        }
+        CHIP_ERROR SetEventPaths(const ObjectList<EventPathParams> * pEventPathList)
+        {
+            mEventPaths.Free();
+            const ObjectList<EventPathParams> * eventPath = pEventPathList;
+            size_t eventPathCount                         = 0;
+            while (eventPath)
+            {
+                eventPathCount++;
+                eventPath = eventPath->mpNext;
+            }
+            ReturnErrorCodeIf((eventPathCount * sizeof(EventPathParamsValues)) > UINT16_MAX, CHIP_ERROR_NO_MEMORY);
+            mEventPaths.Calloc(eventPathCount);
+            ReturnErrorCodeIf(mEventPaths.Get() == nullptr, CHIP_ERROR_NO_MEMORY);
+            eventPath = pEventPathList;
+            for (size_t i = 0; i < eventPathCount; i++)
+            {
+                mEventPaths[i].SetValues(eventPath->mValue);
+                eventPath = eventPath->mpNext;
+            }
+            return CHIP_NO_ERROR;
+        }
+    };
+
+    using SubscriptionInfoIterator = CommonIterator<SubscriptionInfo>;
+
+    virtual ~SubscriptionResumptionStorage(){};
+
+    /**
+     * Iterate through persisted subscriptions
+     *
+     * @return A valid iterator on success. Use CommonIterator accessor to retrieve SubscriptionInfo
+     */
+    virtual SubscriptionInfoIterator * IterateSubscriptions() = 0;
+
+    /**
+     * Save subscription resumption information to storage.
+     *
+     * @param subscriptionInfo the subscription information to save - caller should expect the passed in value is consumed
+     */
+    virtual CHIP_ERROR Save(SubscriptionInfo & subscriptionInfo) = 0;
+
+    /**
+     * Delete subscription resumption information by node ID, fabric index, and subscription ID.
```

Comment:
This should be clearer about this being the _subscriber_ node id.

And I wonder whether this should take ScopedNodeId instead of two separate arguments for node id and index.